### PR TITLE
Convert front end Docker image to use env vars

### PR DESCRIPTION
**Pull request by MrCreosote**
vs. a massive command string which is heinous in Rancher

### CODE REVIEW

File: scripts/fe-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# NOTE: All FE nodes in a cluster must have the same http port:
+# https://docs.starrocks.io/docs/deployment/deploy_manually/
+
+export HTTP_PORT=${HTTP_PORT:-8030}
+export THRIFT_RPC_PORT=${THRIFT_RPC_PORT:-9020}
+export SQL_QUERY_PORT=${SQL_QUERY_PORT:-9030}
+export EDIT_LOG_PORT=${EDIT_LOG_PORT:-9010}
+
+echo "STARROCKS_HOME=${STARROCKS_HOME}"
```
Comment (Tianhao-Gu) [inline scripts/fe-entrypoint.sh:16]: echo for debugging?

File: docker-compose.yml [hunk 2]
```diff
@@ -42,40 +42,30 @@ services:
...
-        echo "# the details provided above" >> /opt/starrocks/fe/conf/fe.conf
-        echo "enable_load_volume_from_conf = true" >> /opt/starrocks/fe/conf/fe.conf
-
-        /opt/starrocks/fe/bin/start_fe.sh
+    environment:
+      - STARROCKS_HOME=/starrocks/fe1
+      - HTTP_PORT=8031
+      - THRIFT_RPC_PORT=9021
+      - SQL_QUERY_PORT=9031
+      - EDIT_LOG_PORT=9011
```
Comment (Tianhao-Gu) [inline docker-compose.yml:56]: why not just using the default ports assigned by the entrypoint?

Comment (Tianhao-Gu) [reply docker-compose.yml:56]: Or should we not assign default ports for those in the entrypoint.sh and exit if they are not set?

Comment (MrCreosote) [reply docker-compose.yml:56]: This is just an example for how to specify the ports. If you want default ports, delete / comment out these lines

File: scripts/fe-entrypoint.sh (hunk 1 above)
Comment (MrCreosote) [reply scripts/fe-entrypoint.sh:16]: That and makes it easy to check logs if you forget ports

Comment (Tianhao-Gu) [reply scripts/fe-entrypoint.sh:16]: 👍

File: docker-compose.yml (hunk 2 above)
Comment (Tianhao-Gu) [reply docker-compose.yml:56]: 👍

**Review by Tianhao-Gu — APPROVED** [review]
👍

**Pull request merged**